sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -165,10 +180,17 @@ gnrc_pktsnip_t *_dio_prefix_info_build(gnrc_pktsnip_t *pkt, gnrc_rpl_dodag_t *do
     prefix_info->length = GNRC_RPL_OPT_PREFIX_INFO_LEN;
     /* auto-address configuration */
     prefix_info->LAR_flags = GNRC_RPL_PREFIX_AUTO_ADDRESS_BIT;
-    /* TODO: Get real values from NIB_PL */
-    prefix_info->valid_lifetime = UINT32_MAX;
-    prefix_info->pref_lifetime = UINT32_MAX;
     prefix_info->prefix_len = 64;
+    if (_get_pl_entry(dodag->iface, &dodag->dodag_id, prefix_info->prefix_len,
+                      &ple)) {
+        prefix_info->valid_lifetime = byteorder_htonl(ple.valid_until);
+        prefix_info->pref_lifetime = byteorder_htonl(ple.pref_until);
```

Comment:
There isn't a (functioning) way to configure this on the dodag root node yet, is there?

Reply:
Yes there is: Add the GUA to the interface, use `gnrc_ipv6_nib_pl_set()` to update the lifetimes. There you go (everything of this can also be done using the shell).

Reply:
> Add the GUA to the interface, use gnrc_ipv6_nib_pl_set() to update the lifetimes. There you go (everything of this can also be done using the shell).

Thanks, should have noticed that this is possible :)

According to [rfc 6550, 6.7.10](https://tools.ietf.org/html/rfc6550#section-6.7.10) valid and pref lifetime are in seconds while the prefix lifetimes are in milliseconds. Furthermore, if I understand the code correctly (It's getting late already) `*_until` is defined as the system time until the prefix is valid. I think this should become something like:
```
prefix_info->valid_lifetime = byteorder_htonl((ple.valid_until - now)/1000);
prefix_info->pref_lifetime = byteorder_htonl((ple.pref_until - now)/1000);
```

Reply:
Oops, true. Fixed!